sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -31,16 +33,25 @@ const paths = {
 export class TestProxyHttpClient {
   private url = "http://localhost:5000";
   public recordingId?: string;
-  public mode: string;
-  public httpClient: HttpClient;
+  public mode!: string;
```

Comment:
It's a code smell that these assertions have to be made.  Based on reading below, it seems a little weird that basically every one of the methods is wrapped in an if statement. It leads you to situations where nothing is definitely assigned, so you have to make these exclamation mark type assertions, and that may be fine but defeats a lot of the compiler's assistance.

Reply:
removing the exclamations. 🙂